### ISSUE TITLE
Grpc code generation disregards include and  exclude

### ISSUE THREAD

**Issue by agreedSkiing** [quarkusio/quarkus#41225]
### Describe the bug

Whilst generating a part of the [external-authz](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/ext_authz/v3/ext_authz.proto) project from the io.envoyproxy.controlplane:api dependency I noticed that it generated  **3472 source files** and that is with `scan-for-proto-include` and `scan-for-proto-exclude` set (see below)
```properties
# quarkus.generate-code.grpc.scan-for-imports=none
quarkus.generate-code.grpc.scan-for-proto=io.envoyproxy.controlplane:api
quarkus.generate-code.grpc.scan-for-proto-include."io.envoyproxy.controlplane:api"=envoy/service/auth/v3/external_auth.proto
quarkus.generate-code.grpc.scan-for-proto-exclude."io.envoyproxy.controlplane:api"=**/*v2*/*.*
```

But if I extract the files from io.envoyproxy.controlplane:api for the external_auth.proto then it creates **156 source files**
```properties
# quarkus.generate-code.grpc.scan-for-imports=none
quarkus.generate-code.grpc.scan-for-proto=io.envoyproxy.controlplane:api
quarkus.generate-code.grpc.scan-for-proto-include."io.envoyproxy.controlplane:api"=envoy/service/auth/v3/external_auth.proto
quarkus.generate-code.grpc.scan-for-proto-exclude."io.envoyproxy.controlplane:api"=**/*v2*/*.*
```

### Expected behavior

Generating only needed files for the specified protos.

### Actual behavior

Generates more files than expected

### How to Reproduce?

[external-authorization-grpc.zip](https://github.com/user-attachments/files/15848187/external-authorization-grpc.zip)

### Output of `uname -a` or `ver`

Linux amd-burk 6.6.32-1-MANJARO #1 SMP PREEMPT_DYNAMIC Mon May 27 03:41:25 UTC 2024 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "17.0.2" 2022-01-18 OpenJDK Runtime Environment (build 17.0.2+8-86) OpenJDK 64-Bit Server VM (build 17.0.2+8-86, mixed mode, sharing)

### Quarkus version or git rev

   <quarkus.platform.version>3.11.2</quarkus.platform.version>

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.6 (84538c9988a25aec085021c365c560670ad80f63) Maven home: /home/eriase/.m2/wrapper/dists/apache-maven-3.8.6-bin/67568434/apache-maven-3.8.6 Java version: 17.0.2, vendor: Oracle Corporation, runtime: /home/eriase/.asdf/installs/java/openjdk-17.0.2 Default locale: en_US, platform encoding: ANSI_X3.4-1968 OS name: "linux", version: "6.6.32-1-manjaro", arch: "amd64", family: "unix"

**Comment by cescoffier** [comment]
Yes, that's a change in protoc which makes computing the command line way more complicated.
Any plain usage of protoc with the right command is more than welcome.

**Comment by cescoffier** [comment]
@agreedSkiing Do you know the _bare protoc_ command that does the include as you expect them to be?

**Comment by agreedSkiing** [comment]
I do not know the protoc command for that :(, this issue is mostly based on an assumption on how it should be generated. 

The only thing I can provide is the zip attached in the issue which uses the maven profile _ext-authz_ and _external-auth_ to generate the needed files to my knowledge for this server, the _specified_ profile is trying to do the same thing as the profile _ext-authz_ or _external-auth_ (I don't really remember since I haven't been able to use my account in a while).

Either how the maven profile _external-auth_ uses `<quarkus.grpc.codegen.proto-directory> ${project.basedir}/protos/external-auth </quarkus.grpc.codegen.proto-directory>` and _ext-authz_ uses `<quarkus.grpc.codegen.proto-directory> ${project.basedir}/protos/external-auth </quarkus.grpc.codegen.proto-directory>` which are set in the pom.xml

**Comment by agreedSkiing** [comment]
The other profiles, _exclude_ and _include_, where to test if i could just generate parts of the proto files from the `io.envoyproxy.controlplane:api` dependecy

**Comment by cescoffier** [comment]
Can you attach the zip, maybe using something smaller than envoy.

**Comment by agreedSkiing** [comment]
Will see what I can do :)

**Comment by agreedSkiing** [comment]
Gave it a try with just the google/api/http.proto file instead and got simliar results. It is the same project but with a updated README.md, application.properties, pom.xml and quarkus version.

[external-authorization-grpc.tar.gz](https://github.com/user-attachments/files/18547035/external-authorization-grpc.tar.gz)

**Comment by agreedSkiing** [comment]
Since I started tinkering yesterday I gave it another try today. Streamlined the config to just include a more standard pom.xml and used the `-Dquarkus.profile` to activate different profiles. 

Well the same result as yesterday :(, so 3435 sources files where generated from the io.envoyproxy.controlplane:api dependecy each time I tried to use a include-one-file-v* profile that tried to only include the `google/api/http.proto` (https://github.com/envoyproxy/java-control-plane/blob/v1.0.45/api/src/main/proto/google/api/http.proto) and before that i even tried to exclude all files with my exclude-all-v* profiles but that also generated 3435 sources files :(.

Attached the version of the project also

[grpc-plain-text-quickstart.tar.gz](https://github.com/user-attachments/files/18549654/grpc-plain-text-quickstart.tar.gz)

**Comment by cescoffier** [comment]
Thanks, we will have a look.

@alesj fancy some protoc command hacks?
